sqlglot/generator.py

```diff
@@ -2589,6 +2594,28 @@ def safedpipe_sql(self, expression: exp.SafeDPipe) -> str:
         return self.dpipe_sql(expression)
 
     def div_sql(self, expression: exp.Div) -> str:
+        if self.TYPED_DIVISION:
+            l, r = expression.left, expression.right
+
+            if not l.is_type(*exp.DataType.FLOAT_TYPES) and not r.is_type(
+                *exp.DataType.FLOAT_TYPES
+            ):
+                l.replace(exp.cast(l.copy(), to=exp.DataType.Type.DOUBLE))
+
+        return self.binary(expression, "/")
+
+    def typeddiv_sql(self, expression: exp.TypedDiv) -> str:
+        if self.TYPED_DIVISION:
+            return self.binary(expression, "/")
+
+        l, r = expression.left, expression.right
+        if l.is_type(*exp.DataType.INTEGER_TYPES) and r.is_type(*exp.DataType.INTEGER_TYPES):
```

Comment:
Nit: maybe we could negate the condition here and just do a single `return self.binary(expression, "/")`?

Reply:
Yeah, I thought about that..
That's less code, but I think it's slightly harder to read.
Negates conditions are generally confusing. Plus, this keeps the branch symmetrical with the div_sql

Reply:
Sounds good, marking this as resolved then 👍